src/mokka/src/integration-test/java/pl/hycom/mokka/emulator/mock/MockConfigurationControllerIT.java

```diff
@@ -0,0 +1,180 @@
+    public void request_should_add_MockConfiguration_to_DB_and_WireMockMapping() throws Exception {
+
+        String requestBody = "{\"editMode\":true,\"isNew\":true,\"detailMode\":true,\"showDetails\":false,\"status\":\"" + status + "\",\"httpMethod\":\"" + httpMethod + "\",\"path\":\"" + path + "\",\"name\":\"" + name + "\",\"type\":\"string\",\"configurationContent\":{\"string\":{\"value\":\"test\"}}}\n";
+
+        mvc.perform(asyncRequest(put("/configuration")).contentType(MediaType.APPLICATION_JSON_VALUE).with(csrf())
+        .content(requestBody))
+        .andExpect(status().isOk());
+
+        Response response = RestAssured.get("/__admin/mappings");
+        expect().statusCode(200).header(CONTENT_TYPE_KEY, MediaType.APPLICATION_JSON_VALUE).response();
+        String mappings = response.getBody().jsonPath().get("mappings").toString();
+        UUID expectedUUID = UUID.nameUUIDFromBytes(mockConfigurationRepository.findAll().iterator().next().getId().toString().getBytes());
+
+        Assertions.assertTrue(mappings.contains(expectedUUID.toString()));
+        Assertions.assertEquals(1, StreamSupport.stream(mockConfigurationRepository.findAll().spliterator(), false).count());
+    }
+
+    @Test
+    @Order(2)
+    @WithMockUser(roles="ADMIN")
+    public void request_should_edit_MockConfiguration_on_DB_and_WireMockMapping() throws Exception {
+
+        String id = mockConfigurationRepository.findAll().iterator().next().getId().toString();
+        String testChange = "test_change";
+
+        String requestBody = "{\"editMode\":true,\"isNew\":false,\"detailMode\":true,\"showDetails\":true,\"status\":" + status + ",\"httpMethod\":\"" + httpMethod + "\",\"path\":\""+path+"\",\"id\":"+id+",\"description\":\"test\",\"pattern\":null,\"name\":\"" + testChange + "\",\"enabled\":true,\"proxyBaseUrl\":null,\"timeout\":0,\"order\":0,\"configurationContent\":{\"string\":{\"value\":\"d\"}},\"updated\":\"2020-03-18T01:44:39.089+0000\",\"errors\":null,\"changes\":null,\"showChanges\":false,\"showConfiguration\":true,\"type\":\"string\"}";
+
+        mvc.perform(asyncRequest(put("/configuration")).contentType(MediaType.APPLICATION_JSON_VALUE).with(csrf())
+            .content(requestBody))
+            .andExpect(status().isOk());
+
+        Response response = RestAssured.get("/__admin/mappings");
+        expect().statusCode(200).header(CONTENT_TYPE_KEY, MediaType.APPLICATION_JSON_VALUE).response();
+        String mappings = response.getBody().jsonPath().get("mappings").toString();
+        UUID expectedUUID = UUID.nameUUIDFromBytes(mockConfigurationRepository.findAll().iterator().next().getId().toString().getBytes());
+
+        Assertions.assertTrue(mappings.contains(expectedUUID.toString()));
+        Assertions.assertTrue(mappings.contains(testChange));
+        Assertions.assertEquals(1, StreamSupport.stream(mockConfigurationRepository.findAll().spliterator(), false).count());
+    }
+
+    @Test
+    @Order(3)
```

Comment:
To be removed. Test should setup data on its own. Just make a put at the very beginning that will create MockConfiguration.